ctlog/ctlog.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package ctlog contains data structures and methods to do with CT Log metadata
+// that is needed by the monitor.
+//
+// TODO(katjoyce): Try to come up with a better package name.
+package ctlog
+
+import (
+	"crypto"
+	"fmt"
+
+	ct "github.com/google/certificate-transparency-go"
+	"github.com/google/certificate-transparency-go/logid"
+)
+
+// Log represents a CT Log and contains the Log metadata needed by the monitor.
+type Log struct {
+	ID        logid.LogID
+	Name      string
+	URL       string
+	PublicKey crypto.PublicKey
+}
+
+// New creates a Log structure, populating the fields appropriately.
+//
+// TODO(katjoyce): replace this implementation with something less hacky that
+// takes log details from a log list struct based on the new Log list JSON
+// schema.
+func New(url, name, b64PubKey string) (*Log, error) {
+	id, err := logid.FromPubKeyB64(b64PubKey)
+	if err != nil {
+		return nil, fmt.Errorf("logid.FromPubKeyB64(): %s", err)
```

Comment:
These should probably be %v. Also might not be necessary to wrap the errors if they're unambiguous from logid. Add the formatting there and pass them through here is cleaner.

Reply:
OK but I still think it's better to avoid wrapping errors as we pass GRPC status codes around a lot.

Reply:
I don't really understand this comment.... this project doesn't use GRPC...

I think there can be benefit to wrapping errors to add more context to what cause them, particularly if they could be bubbled up through a lot of layers, but you're right that here it doesn't really add much, so have removed it and returned the errors directly.

Reply:
I meant that in Go unlike e.g. Java this is replacing the error not encapsulating it. If there's any additional information in the specific error type it doesn't get propagated. Code in our other repos uses the status codes in non-GRPC code to indicate the category what went wrong in a reliable way via status.Code(err). That doesn't mean you should do this here, it's up to you if it makes sense or not.

Reply:
Oh I see :)  yes that makes more sense now!